docs/aws/audit/dynamodbmonitoring/rules/kinesis_stream_encrypted_with_cmk.mdx

```diff
@@ -23,6 +23,218 @@ ISO27001, HIPAA
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access Kinesis Stream and DynamoDB.
+
+2. Once AWS CLI is installed and configured, you can use the following command to list all the Kinesis streams:
+
+   ```
+   aws kinesis list-streams
+   ```
+
+   This command will return a list of all the Kinesis streams in your AWS account.
+
+3. To check if a specific Kinesis stream is encrypted with a CMK, you can use the following command:
+
+   ```
+   aws kinesis describe-stream --stream-name your-stream-name
+   ```
+
+   Replace 'your-stream-name' with the name of your Kinesis stream. This command will return a JSON object that contains information about the stream.
+
+4. In the returned JSON object, look for the 'StreamDescription' field. Inside this field, there should be a 'StreamEncryptionType' field. If the value of this field is 'KMS', then the stream is encrypted with a CMK. If the value is 'NONE', then the stream is not encrypted. If the 'StreamEncryptionType' field is not present, then the stream is not encrypted.
+
+   Here is an example of what the 'StreamDescription' field might look like for a stream that is encrypted with a CMK:
+
+   ```
+   "StreamDescription": {
+       ...
+       "StreamEncryptionType": "KMS",
+       ...
+   }
+   ```
+
+   If the 'StreamEncryptionType' field is not present or if its value is 'NONE', then the stream is not encrypted.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials:
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+Replace 'YOUR_ACCESS_KEY' and 'YOUR_SECRET_KEY' with your actual AWS access key and secret key.
+
+3. Now, create a Kinesis client and list all the Kinesis streams:
+```python
+kinesis = session.client('kinesis')
+streams = kinesis.list_streams()
+```
+
+4. For each stream, describe the stream and check if it's encrypted with a CMK:
+```python
+for stream_name in streams['StreamNames']:
+    stream_description = kinesis.describe_stream(StreamName=stream_name)
+    if 'StreamEncryptionType' in stream_description['StreamDescription']:
+        if stream_description['StreamDescription']['StreamEncryptionType'] == 'KMS':
+            print(f"Stream {stream_name} is encrypted with a CMK")
+        else:
+            print(f"Stream {stream_name} is not encrypted with a CMK")
+    else:
+        print(f"Stream {stream_name} does not have encryption enabled")
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>